pavelib/utils/test/suites/pytest_suite.py

```diff
@@ -142,17 +142,30 @@ def cmd(self):
         if self.disable_capture:
             cmd.append("-s")
 
-        if self.processes == -1:
-            cmd.append('-n auto')
-            cmd.append('--dist=loadscope')
-        elif self.processes != 0:
-            cmd.append('-n {}'.format(self.processes))
+        if self.xdist_ip_addresses:
             cmd.append('--dist=loadscope')
-
-        if not self.randomize:
-            cmd.append('-p no:randomly')
-        if self.eval_attr:
-            cmd.append("-a '{}'".format(self.eval_attr))
+            for ip in self.xdist_ip_addresses.split(' '):
+                xdist_string = '--tx ssh=ubuntu@{}//python="source /edx/app/edxapp/edxapp_env; ' \
+                               'python"//chdir="/edx/app/edxapp/edx-platform"'.format(ip)
+                cmd.append(xdist_string)
+            already_synced_dirs = set()
+            for test_path in self.test_id.split():
+                test_root_dir = test_path.split('/')[0]
+                if test_root_dir not in already_synced_dirs:
+                    cmd.append('--rsyncdir {}'.format(test_root_dir))
+                    already_synced_dirs.add(test_root_dir)
+        else:
+            if self.processes == -1:
+                cmd.append('-n auto')
+                cmd.append('--dist=loadscope')
+            elif self.processes != 0:
+                cmd.append('-n {}'.format(self.processes))
+                cmd.append('--dist=loadscope')
+
+            if not self.randomize:
+                cmd.append('-p no:randomly')
+            if self.eval_attr:
+                cmd.append("-a '{}'".format(self.eval_attr))
```

Comment:
While we don't strictly need this in the xdist case (we've mainly been using it for sharding so far), there's nothing preventing them from being used together to run a marked subset of tests in parallel.  So this may also be a candidate for moving out of this `else` block.